File: remove_bg.py
import sys
try:
    from PIL import Image  # type: ignore
    import numpy as np  # type: ignore
except ImportError:
    import subprocess
    subprocess.check_call([sys.executable, '-m', 'pip', 'install', 'Pillow', 'numpy'])
    from PIL import Image  # type: ignore
    import numpy as np  # type: ignore

def remove_bg():
    img = Image.open('public/icon.png').convert('RGBA')
    data = np.array(img)
    r, g, b, a = data[:,:,0], data[:,:,1], data[:,:,2], data[:,:,3]
    
    threshold = 240
    white_mask = (r > threshold) & (g > threshold) & (b > threshold)
    data[white_mask, 3] = 0
    
    for t in range(200, 241):
        partial_mask = (r > t) & (g > t) & (b > t) & ~white_mask
        factor = (255 - t) / (255 - 200)
        data[partial_mask, 3] = (data[partial_mask, 3] * factor).astype(np.uint8)
        
    result = Image.fromarray(data)
    result.save('public/icon_transparent.png', 'PNG')
    print('Done!')

remove_bg()

File: remove_bg.js
import { Jimp } from 'jimp';

async function removeBg() {
  try {
    const image = await Jimp.read('public/icon.png');
    const data = image.bitmap.data;
    
    for (let i = 0; i < data.length; i += 4) {
      const r = data[i];
      const g = data[i+1];
      const b = data[i+2];
      const a = data[i+3];
      
      let newAlpha = a;
      
      if (r > 240 && g > 240 && b > 240) {
        newAlpha = 0;
      } else if (r > 200 && g > 200 && b > 200) {
        const maxColor = Math.max(r, g, b);
        const factor = (255 - maxColor) / 55;
        newAlpha = Math.round(a * factor);
      }
      
      data[i+3] = newAlpha;
    }
    
    await image.write('public/icon_transparent.png');
    console.log('Background removed successfully.');
  } catch (err) {
    console.error('Error:', err);
  }
}

removeBg();
